lib/PHPCfg/AstVisitor/NameResolver.php

```diff
@@ -81,7 +84,7 @@ protected function parseTypeDecl($type) {
         }
         $regex = '(^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*\\\\)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$)';
         if (!preg_match($regex, $type)) {
-            throw new \RuntimeException("Unknown type declaration found: $type");
+            return false;
```

Comment:
You're absolutely right, my bad. But perhaps the early return for the variable should be merged with this then? https://www.phpdoc.org/docs/latest/references/phpdoc/tags/param.html states that the type MUST be specified, so encountering a variable is a special case of a malformed type.